rllib/utils/actor_manager.py

```diff
@@ -312,28 +362,29 @@ def __fetch_result_and_mark_state(
 
         # Remote data should already be fetched to local object store at this point.
         remote_results = RemoteCallResults()
-        for i, r in enumerate(ready):
-            actor_idx = remote_actor_indices[i]
+        for r in ready:
+            # Find the corresponding actor ID for this remote call.
+            actor_id = remote_actor_ids[remote_calls.index(r)]
```

Comment:
I don't see index method implemented for RemoteCallResults class. How does this work? what's the run time of finding the index? Should `RemoteCallResults()` internally be a dict so that the look up is O(1)?

Reply:
``remote_calls`` is a List[ObjectRef]. index is an API for python ilst.
I think it's likely ok, we will never have millions of outstanding requests, so probably shouldn't complicate the API for performance reasons.
modern computer is pretty fast ... :)

Reply:
got it. makes sense.